docs/data-structures/binary-tree.mdx

```diff
@@ -0,0 +1,802 @@
+        root->left = NULL;
+        root->right = NULL;
+    }
+}
+
+/* inserts the key, such that the Tree follows the definition of a binary tree.
+   the function travels down the tree starting from the root to find a suitable position.
+   if key is less than the key_value at a node, function calls itself with its left subtree.
+   if key is greater than the key_value at a node, function calls itself with its right subtree.
+   When it reaches a suitable place, it creates a new node. This node gets key_value as key and child nodes as NULL*/
+void binaryTree::insertKey(int key, node* leaf){
+    if(key< leaf->key_value){
+        if(leaf->left!=NULL){
+            insertKey(key, leaf->left);
+        }
+        else{
+            leaf->left = new node;
+            leaf->left->key_value = key;
+            leaf->left->left = NULL;
+            leaf->left->right = NULL;
+        }
+    }
+    else if(key>= leaf->key_value){
+        if(leaf->right!=NULL){
+            insertKey(key, leaf->right);
+        }
+        else{
+            leaf->right = new node;
+            leaf->right->key_value = key;
+            leaf->right->left = NULL;
+            leaf->right->right = NULL;
+        }
+    }
+}
+
+// calls preOrderTraversal_recursive, a private member-function
+void binaryTree::preOrderTraversal_recursive(){
+    cout<<"PreOrderTraversal(recursive): ";
+    preOrderTraversal_recursive(root);
+}
+
+/* for a particular node, function prints its key_value and recursively calls
+   its left and right subtree(in that order).*/
+void binaryTree::preOrderTraversal_recursive(node* leaf){
+    if(leaf!=NULL){
+        cout<<leaf->key_value<<" ";
+        preOrderTraversal_recursive(leaf->left);
+        preOrderTraversal_recursive(leaf->right);
+    }
+}
+
+/* If root= NULL, function exits, else: 
+1. Root is set as current node.
+2. key_value of the current node is the output, and address of its right subtree is pushed in stack. Also, sets current node as address of its left subtree. This continues until, current node = NULL.
+3. On reaching NULL i.e. no left subtree is available, it sets current node as top value in the stack and pops it from the stack. Then it repeats step 2.*/
+void binaryTree::preOrderTraversal_iterative(){
+    cout<<"PreOrderTraversal(iterative): ";
+    if(root==NULL)
+        return;
+    stack<node*> nodeAddr;
+    node* currentNode = root;
+
+    while(currentNode!=NULL || !nodeAddr.empty()){
+        while(currentNode!=NULL){
+            cout<<currentNode->key_value<<" ";
+            if(currentNode->right!=NULL){
+                nodeAddr.push(currentNode->right);
+            }
+            currentNode = currentNode->left;
+        }
+        if(!nodeAddr.empty()){
+            currentNode = nodeAddr.top();
+            nodeAddr.pop();
+        }
+    }
+}
+
+int main(){
+    binaryTree Tree;
+    //the elements to be inserted in the tree
+    vector<int> elements = {25,15,50,10,22,35,70,4,12,18,24,31,44,66,80};
```

Comment:
Can you please make it dynamic input? 

By dynamic, I mean that the user has to input their own values rather than having a test case for the same. In rest of the C++ codes, we have dynamic inputs, so it would be good to have them here as well